### DISCUSSION TITLE
Etsy test stores

### DISCUSSION THREAD

**Discussion by missnicbic** [etsy/open-api#871] [Q&A]
Hi all! We are currently onboarding a QA team to help us improve the quality of our Etsy integration however hitting some issues.

Is it possible to create additional development stores for testing under different accounts and what are the current guidelines?

Does anyone have any other solutions other than ngrok for creating oauth endpoint calls so that we can test in UAT and local?

Many thanks!

**Comment by Gareth064** [comment]
There is no such things as a "real" test shop. You just create a shop like you would normally. Then you can put it into Developer Mode. Dev mode just means that any listings on the shop won't be viewable in search results etc.

This is there policy.

https://www.etsy.com/legal/policy/api-testing-policy/169130941112

Regarding ngrok etc. You don't need it. You can add localhost as an acceptable Callback URL so when you go through the OAuth dance, Etsy will allow the use of localhost. There are zero scenarios where Etsy make a request to your app. Its all one way, your make a request, they respond.

**Comment by missnicbic** [comment]
Thanks for this, I went ahead and created a completely new secondary account under a different login and found that the Developer Mode setting is only available if you have an app registered to the same account. 

Etsy Trust and Safety then shut it down today, presumably as I don't have this setting on (however have carefully followed all the other rules around the dev account set up, including archiving listings and marking clearly as a test account). 

I'm basically trying to create a situation where my support team and QA team can access an Etsy store for the purposes of testing out and mocking up certain situations with products and orders. Based on the Trust and Safety issue that occurred with the secondary account, I'm not keen to provide my primary account access to my team (and also not keen on sharing credentials). 

The only other solution here is that I create the scenarios myself, however this is something that with Shopify, Square et al we are offloading to our team.

How are others setting up these situations to test, given Etsy doesn't have a sandbox or development environment?

**Comment by missnicbic** [comment]
We just received a reply from Trust and Safety indicating that all of our accounts will be suspended and are deeply concerned. We are a long standing Etsy app of over 10 years, and are also one of the featured apps on your Integrations page (Craftybase). 

We are fine with having the additional test account deleted if this isn't the way to create a secondary test environment, just want to confirm that this suspension is not going to have any impact on our developer account and API keys as this would be a huge problem for us and for the thousands of Etsy sellers that rely on our app.

If someone could respond from the Etsy API team asap that would be much appreciated. @nateswart  @capndesign

**Reply by nateswart** [reply]
I'll follow up privately on this.

**Reply by gil-urshipper** [reply]
@nateswart, I’m also facing the same issue atm. Would you be able to help, please? TIA 🙏

**Comment by Jameslin-9880** [comment]
same issue happened to me, 

they suspended our account and app even we have explicitly added a name: "testing" on the seller account  based on their instruction
